### PR TITLE
Fixed vanishing gradient problem + changed lr_scheduler

### PR DESCRIPTION
**Pull request by Alfred-N**
Fixed vanishing gradient problem + changed CosineAnnealingLR to CosineAnnealingWarmRestarts,

### CODE REVIEW

**Review by YinengWang — APPROVED** [review]
Great! It turns out that sometimes graphs can tell a lot of things.

**Review by rlindsberg — APPROVED** [review]
Really great work with your attention to details and digging deep in the paper!
